### PR TITLE
On openshift, build and deploy from github

### PR DESCRIPTION
**Pull request by henrynash**
On Openshift, we should use the ability to build and deploy directly from GitHub source, rather than have people build their own image using docker. This also prevents issues of ensuring you have access to the IBM Cloud container registry.

It is a pity that `oc new-app...` does not provide an ability to pass in a secret to be exposed via environment variables in the application instance (it does support it for the build instance). Hence the requirement to patch the deployment after the build. I'm open to a better way of doing this (that does not require cloning and modifying the source).

### CODE REVIEW

File: README-OpenShift.md [hunk 1]
```diff
@@ -56,35 +28,47 @@ Example:
...
-oc get pods -l app=get-started-node
+oc expose svc/get-started-node
 ```
 
-2. Identify your LoadBalancer Ingress IP using `oc get service get-started-node`
-3. Access your application at `http://<EXTERNAL-IP>:8080/`
+## Access the application
+
+1. Identify the route to the application `oc get route get-started-node`
+2. Access your application via the URL given in the HOST/PORT column, which will be of the form `get-started-node-.....containers.appdomain.cloud`
```
Comment (hickeyma) [inline README-OpenShift.md:63]: Can you explicitly specify that it is HTTP and NOT HTTPS for the URL?

File: README-OpenShift.md [hunk 2]
```diff
@@ -56,35 +28,47 @@ Example:
...
 
+If, and when, you want to remove the application:
+
 ```shell
-oc delete deployment,service -l app=get-started-node
+oc delete deployment,route -l app=get-started-node
 oc delete secret cloudant
 ```
+
+You should also delete any build pods that were generated by building from source.
```
Comment (hickeyma) [inline README-OpenShift.md:74]: Would it be worth adding a link/instructions on how to do this?

**Review by hickeyma — CHANGES_REQUESTED** [review]
I like this as it uses the build capabilities in OpenShift to deploy easily. Thanks for pushing these changes @henrynash. Just few small things to address.

File: README-OpenShift.md (hunk 1 above)
Comment (henrynash) [reply README-OpenShift.md]: Added comment to clarify this is HTTP.

File: README-OpenShift.md (hunk 2 above)
Comment (henrynash) [reply README-OpenShift.md]: Add command to do this.

**Comment by henrynash** [comment]
Thanks your review @hickeyma, fixed up accordingly.

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @henrynash

**Comment by hickeyma** [comment]
@kevin-ortega Do you mind reviewing this PR when you get a chance?

**Review by kevin-ortega — APPROVED** [review]

**Comment by hickeyma** [comment]
Thanks @kevin-ortega

**Pull request merged**